### ISSUE TITLE
Wrong rewrap on lists in Markdown / AsciiDocs

### ISSUE THREAD

**Issue by n2o** [stkb/Rewrap#124]
Hello,
thanks for this great plugins which serves very well when working with VSCode. But apparently there is one problem, which causes me some headaches and prevents me from using Rewrap on Markdown / Asciidoc-Lists. Here is a description of the problem. Thanks for your support!

#### Problem

When trying to wrap a list containing more than the specified numbers of characters, it puts all list-items behind each other and creates a long string, which is then being wrapped. This breaks the definition of a list.

#### Sample
Given the following list in a markdown / asciidoc sample:

```markdown
1. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx
2. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx 
3. xxxxx xxxxx xxxxx xxxxx
```
Place the cursor somewhere on one of the list items and hit <kbd>Alt-Q</kbd> to rewrap the lines produces this result:

```markdown
1. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx
xxxxx xxxxx xxxxx 2. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx
xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx 3. xxxxx xxxxx xxxxx
xxxxx
```
**Expected Output**
```markdown
1. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx 
   xxxxx xxxxx xxxxx xxxxx 
2. xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx 
   xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx 
3. xxxxx xxxxx xxxxx xxxxx
```

#### Versions
Rewrap: 1.9.1
Visual Studio Code Version: 1.30.2
Commit: 61122f88f0bf01e2ac16bdb9e1bc4571755f5bd8
Date: 2019-01-07T22:49:48.319Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
OS: Linux x64 4.20.0-arch1-1-ARCH

**Comment by NorthboundTrain** [comment]
Just came here to report the same issue. Great plugin otherwise -- thank you!

**Comment by anothernode** [comment]
Seems to work fine for Markdown by now. But would be great to have better AsciiDoc-Support, too.

**Comment by anatolebeuzon** [comment]
Issue is still present for me (v1.14.0)

**Comment by zerok** [comment]
@anothernode Is this still working for you in Markdown files? Here on 1.16.0 I get the behavior that @n2o mentioned.

**Comment by stkb** [comment]
For AsciiDoc support, see #139.

@zerok Could you give a minimal example if what's not working for you? and also the file type & extension. For me, the original list sample is working fine in an empty `.md` file in v1.16.0

**Comment by zerok** [comment]
@stkb Argh sorry, looks like I was by accident in a `.mdx` file (where I'm using [MDX](https://marketplace.visualstudio.com/items?itemName=silvenon.mdx)). In a normal Markdown file everything works 😄

**Comment by stkb** [comment]
@zerok Thanks, I'll publish a fix for .mdx files tomorrow

**Comment by zerok** [comment]
@stkb WOW! Thank you so much!!! ❤️

**Comment by stkb** [comment]
@zerok Just to be clear, the fix just treats .mdx the same as normal markdown files. No special treatment for the javascript parts yet :)

But it's now available in [pre-release v17.3](https://github.com/stkb/Rewrap/releases/tag/beta) (will also be in the next minor stable version).

I'm also going to close this issue now

**Comment by zerok** [comment]
> @zerok Just to be clear, the fix just treats .mdx the same as normal markdown files. No special treatment for the javascript parts yet :)
> 
> But it's now available in [pre-release v17.3](https://github.com/stkb/Rewrap/releases/tag/beta) (will also be in the next minor stable version).
> 
> I'm also going to close this issue now

That's absolutely fine. Thank you so much!
